src/consensus/params.cpp

```diff
@@ -218,12 +258,113 @@ namespace Consensus {
         vFundingStreams[idx] = FundingStream::ParseFundingStream(*this, keyConstants, startHeight, endHeight, strAddresses);
     };
 
-    FundingStreamAddress FundingStream::RecipientAddress(const Consensus::Params& params, int nHeight) const
+    void Params::AddZIP207LockboxStream(
+        const KeyConstants& keyConstants,
+        FundingStreamIndex idx,
+        int startHeight,
+        int addressChangeInterval,
+        int intervalCount)
+    {
+        std::vector<FundingStreamRecipient> recipients(intervalCount, Lockbox());
+        auto validationResult = FundingStream::ValidateFundingStream(
+                *this,
+                startHeight,
+                startHeight + (intervalCount * addressChangeInterval),
```

Comment:
```suggestion
                startHeight + duration,
```

Reply:
Not a correct change, as described above. Ideally the original code would also take a start height and interval count instead of an explicit start and end as well.